app/src/main/java/com/example/atheneum/models/GoodreadsReviewAdapter.java

```diff
@@ -0,0 +1,244 @@
+package com.example.atheneum.models;
+
+import android.util.Patterns;
+import android.util.Xml;
+
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Adapter class for converting the XML response from goodreads into a GoodreadsReveiwInfo Object.
+ * Most of the XML handling taken from the Google tutorial found at
+ * https://developer.android.com/training/basics/network-ops/xml
+ */
+public class GoodreadsReviewAdapter {
+    private String responseXML;
+    private GoodreadsReviewInfo reviewInfo = null;
+
+    /**
+     * Constructor for the object
+     * @param xml the XML response string
+     */
+    public GoodreadsReviewAdapter(String xml) {
+        this.responseXML = xml;
+        try {
+            this.reviewInfo = xmlToReviewInfo();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
You should log the stacktrace instead of printing it so it's easier to trace the source of errors: https://stackoverflow.com/a/3855657/11039833.

Reply:
Fixed